pd/src/components/shielded_pool.rs

```diff
@@ -277,6 +279,33 @@ impl ShieldedPool {
 
     #[instrument(skip(self))]
     async fn write_block(&mut self) -> Result<()> {
+        // Handle any pending reward notes from the Staking component
+        let notes: RewardNotes = self
+            .overlay
+            .get_domain(format!("reward_notes/{}", self.compact_block.height).into())
+            .await?
+            .unwrap_or_default();
```

Comment:
Instead of constructing the path manually here (meaning we have to check that it's constructed consistently), I think we should `use super::staking::View as _;` so that the paths can be in a single place.